From rjw@sisk.pl Mon Mar  3 14:48:06 2008
From: "Rafael J. Wysocki" <rjw@sisk.pl>
Date: Mon, 3 Mar 2008 23:46:51 +0100
Subject: driver core: Remove dpm_sysfs_remove() from error path of device_add()
To: Greg KH <gregkh@suse.de>
Cc: Alan Stern <stern@rowland.harvard.edu>, Len Brown <lenb@kernel.org>, Pavel Machek <pavel@ucw.cz>
Message-ID: <200803032346.52173.rjw@sisk.pl>
Content-Disposition: inline


From: Rafael J. Wysocki <rjw@sisk.pl>

Since device_pm_remove(dev) calls dpm_sysfs_remove(dev), it's
incorrect to call the latter after the former in the device_add()
error path.

Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/core.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -839,7 +839,6 @@ int device_add(struct device *dev)
 	return error;
  BusError:
 	device_pm_remove(dev);
-	dpm_sysfs_remove(dev);
  PMError:
 	if (dev->bus)
 		blocking_notifier_call_chain(&dev->bus->p->bus_notifier,
